blockchain/message_pool/src/msgpool.rs

```diff
@@ -622,21 +595,18 @@ where
                     msg.sequence(),
                     rmsgs.borrow_mut(),
                 )
-                .await;
+                .await?;
             }
         }
-        let mut cur_ts_locked = cur_tipset.write().await;
-        *cur_ts_locked = ts;
-        drop(cur_ts_locked);
+        *cur_tipset.write().await = ts;
     }
     for (_, hm) in rmsgs {
         for (_, msg) in hm {
-            let api_locked = api.clone();
-            let pending = pending.clone();
-            let bls_sig_cache = bls_sig_cache.clone();
-            add_helper(api_locked, bls_sig_cache, pending, msg.clone())
-                .await
-                .ok();
+            if let Err(e) =
+                add_helper(api.clone(), bls_sig_cache.clone(), pending.clone(), msg).await
```

Comment:
so actually you can just change the function signature of this function to take in references so you don't have to clone the `Arc`s, they don't cross any thread bounds